src/magic_enchantment.cpp

```diff
@@ -0,0 +1,453 @@
+    { "STRENGTH", enchantment::mod::STRENGTH },
+    { "DEXTERITY", enchantment::mod::DEXTERITY },
+    { "PERCEPTION", enchantment::mod::PERCEPTION },
+    { "INTELLIGENCE", enchantment::mod::INTELLIGENCE },
+    { "SPEED", enchantment::mod::SPEED },
+    { "ATTACK_COST", enchantment::mod::ATTACK_COST },
+    { "MOVE_COST", enchantment::mod::MOVE_COST },
+    { "METABOLISM", enchantment::mod::METABOLISM },
+    { "MAX_MANA", enchantment::mod::MAX_MANA },
+    { "REGEN_MANA", enchantment::mod::REGEN_MANA },
+    { "BIONIC_POWER", enchantment::mod::BIONIC_POWER },
+    { "MAX_STAMINA", enchantment::mod::MAX_STAMINA },
+    { "REGEN_STAMINA", enchantment::mod::REGEN_STAMINA },
+    { "MAX_HP", enchantment::mod::MAX_HP },
+    { "REGEN_HP", enchantment::mod::REGEN_HP },
+    { "THIRST", enchantment::mod::THIRST },
+    { "FATIGUE", enchantment::mod::FATIGUE },
+    { "PAIN", enchantment::mod::PAIN },
+    { "BONUS_DAMAGE", enchantment::mod::BONUS_DODGE },
+    { "BONUS_BLOCK", enchantment::mod::BONUS_BLOCK },
+    { "BONUS_DODGE", enchantment::mod::BONUS_DODGE },
+    { "ATTACK_NOISE", enchantment::mod::ATTACK_NOISE },
+    { "SPELL_NOISE", enchantment::mod::SPELL_NOISE },
+    { "SHOUT_NOISE", enchantment::mod::SHOUT_NOISE },
+    { "FOOTSTEP_NOISE", enchantment::mod::FOOTSTEP_NOISE },
+    { "SIGHT_RANGE", enchantment::mod::SIGHT_RANGE },
+    { "CARRY_WEIGHT", enchantment::mod::CARRY_WEIGHT },
+    { "CARRY_VOLUME", enchantment::mod::CARRY_VOLUME },
+    { "SOCIAL_LIE", enchantment::mod::SOCIAL_LIE },
+    { "SOCIAL_PERSUADE", enchantment::mod::SOCIAL_PERSUADE },
+    { "SOCIAL_INTIMIDATE", enchantment::mod::SOCIAL_INTIMIDATE },
+    { "ITEM_DAMAGE_BASH", enchantment::mod::ITEM_DAMAGE_BASH },
+    { "ITEM_DAMAGE_CUT", enchantment::mod::ITEM_DAMAGE_CUT },
+    { "ITEM_DAMAGE_STAB", enchantment::mod::ITEM_DAMAGE_STAB },
+    { "ITEM_DAMAGE_HEAT", enchantment::mod::ITEM_DAMAGE_HEAT },
+    { "ITEM_DAMAGE_COLD", enchantment::mod::ITEM_DAMAGE_COLD },
+    { "ITEM_DAMAGE_ELEC", enchantment::mod::ITEM_DAMAGE_ELEC },
+    { "ITEM_DAMAGE_ACID", enchantment::mod::ITEM_DAMAGE_ACID },
+    { "ITEM_DAMAGE_BIO", enchantment::mod::ITEM_DAMAGE_BIO },
+    { "ITEM_DAMAGE_AP", enchantment::mod::ITEM_DAMAGE_AP },
+    { "ITEM_ARMOR_BASH", enchantment::mod::ITEM_ARMOR_BASH },
+    { "ITEM_ARMOR_CUT", enchantment::mod::ITEM_ARMOR_CUT },
+    { "ITEM_ARMOR_STAB", enchantment::mod::ITEM_ARMOR_STAB },
+    { "ITEM_ARMOR_HEAT", enchantment::mod::ITEM_ARMOR_HEAT },
+    { "ITEM_ARMOR_COLD", enchantment::mod::ITEM_ARMOR_COLD },
+    { "ITEM_ARMOR_ELEC", enchantment::mod::ITEM_ARMOR_ELEC },
+    { "ITEM_ARMOR_ACID", enchantment::mod::ITEM_ARMOR_ACID },
+    { "ITEM_ARMOR_BIO", enchantment::mod::ITEM_ARMOR_BIO },
+    { "ITEM_WEIGHT", enchantment::mod::ITEM_WEIGHT },
+    { "ITEM_ENCUMBRANCE", enchantment::mod::ITEM_ENCUMBRANCE },
+    { "ITEM_VOLUME", enchantment::mod::ITEM_VOLUME },
+    { "ITEM_COVERAGE", enchantment::mod::ITEM_COVERAGE },
+};
```

Comment:
Having explicit maps like these should no longer be necessary; the `enum_conversion.h` infrastructure should create them for you automatically.

You will need to specialize `enum_traits` for your enums.